test/e2e/apps/deployment.go

```diff
@@ -785,6 +811,122 @@ func testDeploymentsControllerRef(f *framework.Framework) {
+	Expect(err).NotTo(HaveOccurred(), "error in waiting for pods to come up: %v", err)
+
+	framework.Logf("Waiting for deployment %q to complete", deployment.Name)
+	Expect(framework.WaitForDeploymentComplete(c, deployment)).NotTo(HaveOccurred())
+
+	firstRS, err := deploymentutil.GetNewReplicaSet(deployment, c.ExtensionsV1beta1())
+	Expect(err).NotTo(HaveOccurred())
+
+	// Update the deployment with a non-existent image so that the new replica set will be blocked.
+	framework.Logf("Updating deployment %q with a non-existent image", deploymentName)
+	deployment, err = framework.UpdateDeploymentWithRetries(c, ns, d.Name, func(update *extensions.Deployment) {
+		update.Spec.Template.Spec.Containers[0].Image = "nginx:404"
+	})
+	Expect(err).NotTo(HaveOccurred())
+
+	framework.Logf("Waiting for observed generation %d", deployment.Generation)
+	Expect(framework.WaitForObservedDeployment(c, ns, deploymentName, deployment.Generation)).NotTo(HaveOccurred())
+
+	framework.Logf("Verifying that deployment %q has minimum required number of available replicas", deploymentName)
+	minAvailableReplicas := deploymentutil.MinAvailable(deployment)
+	if deployment.Status.AvailableReplicas < minAvailableReplicas {
+		Expect(fmt.Errorf("observed %d available replicas, less than min required %d", deployment.Status.AvailableReplicas, minAvailableReplicas)).NotTo(HaveOccurred())
+	}
+
+	framework.Logf("Verifying that the replicasets of deployment %q have desired number of replicas", deploymentName)
+	firstRS, err = c.ExtensionsV1beta1().ReplicaSets(ns).Get(firstRS.Name, metav1.GetOptions{})
+	Expect(err).NotTo(HaveOccurred())
+	secondRS, err := deploymentutil.GetNewReplicaSet(deployment, c.ExtensionsV1beta1())
+	Expect(err).NotTo(HaveOccurred())
+
+	firstCond := replicaSetHasDesiredReplicas(c.ExtensionsV1beta1(), firstRS)
+	Expect(wait.PollImmediate(10*time.Millisecond, 1*time.Minute, firstCond)).NotTo(HaveOccurred())
+	secondCond := replicaSetHasDesiredReplicas(c.ExtensionsV1beta1(), secondRS)
+	Expect(wait.PollImmediate(10*time.Millisecond, 1*time.Minute, secondCond)).NotTo(HaveOccurred())
+
+	framework.Logf("Verifying that first rollout's replicaset has minimum required number of available replicas")
+	if firstRS.Status.AvailableReplicas < minAvailableReplicas {
+		Expect(fmt.Errorf("observed %d available replicas, less than min required %d", firstRS.Status.AvailableReplicas, minAvailableReplicas)).NotTo(HaveOccurred())
+	}
+
+	framework.Logf("Verifying that first rollout's replicaset has 8 available replicas")
+	firstCond = replicaSetHasSpecificAvailableReplicas(c.ExtensionsV1beta1(), firstRS, 8)
+	err = wait.PollImmediate(10*time.Millisecond, 1*time.Minute, firstCond)
+	Expect(err).NotTo(HaveOccurred())
+
+	framework.Logf("Verifying that second rollout's replicaset does not have any available replicas")
+	if secondRS.Status.AvailableReplicas != 0 { // pooling should not be needed
```

Comment:
The comment here isn't necessary. Remove it.

Reply:
Use the test format: `Expect` the number not to be `0`

Reply:
+1